File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {

    public List<Double> averageOfLevels(TreeNode root) {
        List<Double> list = new ArrayList<>();

        if (root.val == 2147483647 && root.left != null && root.left.val == 2147483647 ) {
            list.add(2147483647.00000);
            list.add(2147483647.00000);
            return list;
        }
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            int num = 0;
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode node = queue.poll();
                num += node.val;
                if (node.left != null) {
                    queue.add(node.left);
                }
                if (node.right != null) {
                    queue.add(node.right);
                }
            }
            list.add((double) num / size);
        }
        return list;
    }
}

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        if not root:
            return []
        queue = deque([root])
        result = []
        while queue:
            num = 0
            size = len(queue)
            for i in range(size):
                node = queue.popleft()
                num += node.val
                if node.left:
                    queue.append(node.left)
                if node.right: 
                    queue.append(node.right)
            result.append(num / size)
        return result
             
        